model/API.inc.php

```diff
@@ -1323,5 +1323,71 @@ public static function getSearchParamValues($params, $param) {
 		
 		return $sets;
 	}
+
+	public static function getRequestLimits($params) {
+		$rate = null;
+		$concurrent = null;
+
+		if(Z_CONFIG::$REDUCE_LOAD) {
+			if ($params['userID']) {
+				$rate = [
+					'logOnly' => false,
+					'bucket' => $params['userID'] . '_' . $params['ip'],
+					'limit' => 1, //request per second
+					'burst' => 3, //burst multiplier for 'limit'
+					'warn' => 1 //remaining requests threshold to warn client
+				];
+
+				$concurrent = [
+					'logOnly' => false,
+					'bucket' => $params['userID'],
+					'ttl' => 10,
+					'limit' => 2,
+					'warn' => 1
+				];
+			} else {
+				$rate = [
+					'logOnly' => false,
+					'bucket' => $params['ip'],
+					'limit' => 10,
+					'burst' => 1,
+					'warn' => 50
+				];
+			}
+		} else {
+			if ($params['userID']) {
+				$rate = [
+					'logOnly' => false,
+					'bucket' => $params['userID'] . '_' . $params['ip'],
+					'limit' => 1,
+					'burst' => 3,
+					'warn' => 1
+				];
+
+				$concurrent = [
+					'logOnly' => false,
+					'bucket' => $params['userID'],
```

Comment:
Worth keeping in mind that someone could easily have Zotero running on multiple computers, and they'll all pull down changes automatically at the same time, so at the very least this limit should be 2 or 3 times the maximum concurrent requests a client makes (which is probably ~5). That said, as we discussed, a 429 isn't really a problem, particularly for background auto-syncs.